Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2609 @@
+                currentExtentEndTimeLabel.isHidden = true
+            case .thumbs:
+                //
+                // Show current extent labels
+                // only if slider is visible.
+                if isSliderVisible {
+                    currentExtentStartTimeLabel.isHidden = false
+                    currentExtentEndTimeLabel.isHidden = isRangeEnabled ? false : true
+                    updateCurrentExtentLabelFrames()
+                }
+            }
+            
+            // All above cases require to remove
+            // existing tick marks and reposition
+            // them (none will not add it)
+            removeTickMarkLabels()
+            positionTickMarks()
+        }
+    }
+    
+    // MARK: Thumb Properties
+    
+    /**
+     The color used to tint the thumbs.
+     */
+    public var thumbFillColor = UIColor.white {
+        didSet {
+            lowerThumbLayer.setNeedsDisplay()
+            upperThumbLayer.setNeedsDisplay()
+        }
+    }
+    
+    /**
+     The color used to tint the thumb's border.
+     */
+    public var thumbBorderColor = UIColor.black {
+        didSet {
+            lowerThumbLayer.setNeedsDisplay()
+            upperThumbLayer.setNeedsDisplay()
+        }
+    }
+    
+    /**
+     The border width of the thumb.
+     */
+    public var thumbBorderWidth: CGFloat = 1.0 {
+        didSet {
+            lowerThumbLayer.setNeedsDisplay()
+            upperThumbLayer.setNeedsDisplay()
+        }
+    }
+    
+    /**
+     The size of the thumb.
+     */
+    public var thumbSize = CGSize(width: 25.0, height: 25.0) {
+        didSet {
+            //
+            // More than 50 width and height
+            // is too big for the iOS control.
+            // So, restrict it.
+            if thumbSize.width > maximumThumbSize {
+                thumbSize.width = maximumThumbSize
+            }
+            if thumbSize.height > maximumThumbSize {
+                thumbSize.height = maximumThumbSize
+            }
+            
+            // Set side padding of the track layer for the new thumb size
+            // so thumb has enough space to go all the way to end.
+            trackLayerSidePadding = (thumbSize.width / 2.0) + labelSidePadding
+            
+            // Refresh the view.
+            refresh()
+        }
+    }
+    
+    /**
+     The corner radius of the thumb.
+     */
+    public var thumbCornerRadius = 1.0 {
+        didSet {
+            lowerThumbLayer.setNeedsDisplay()
+            upperThumbLayer.setNeedsDisplay()
+        }
+    }
+    
+    // MARK: Playback Properties
+    
+    /**
+     A Boolean value indicating whether playback buttons are visible or not. Default is true.
+     */
+    public var playbackButtonsVisible = true {
+        didSet {
+            layoutIfNeeded()
+            UIView.transition(with: self, duration: 0.5, options: .transitionCrossDissolve, animations: {
```

Comment:
@njarecha You might consider not implicitly animating changes to the time slider's properties. The user may not want them animated, or they may want to use a different duration, or they may want to specify a completion closure.

Reply:
Removed animation